hedera-node/hedera-smart-contract-service-impl/src/main/java/module-info.java

```diff
@@ -6,19 +6,18 @@
     requires transitive com.hedera.node.config;
     requires transitive com.hedera.node.hapi;
     requires transitive com.hedera.pbj.runtime;
+    requires transitive com.swirlds.common;
+    requires transitive com.swirlds.config;
```

Comment:
What does this service expose in its API that requires config?
With the amount of transitives we're adding, it is becoming important to be sure we actually need transitive for all of them.

Reply:
I did the new dependency analysis – `checkAllModuleInfo` – on this branch and addressed all things reported. I did that to have a "clean" state to check that everything is fine with the changes in the module patching we did.

```
./gradlew app-service-contract-impl:reason --id com.swirlds:swirlds-config 


----------------------------------------
You asked about the dependency 'com.swirlds:swirlds-config-api:0.40.0-adhoc.x5482f0a8'.
There is no advice regarding this dependency.
----------------------------------------

...

Source: main
------------
* Exposes 1 class: com.swirlds.config.api.Configuration (implies api).
```

I believe this is because:
```
com.hedera.node.app.service.contract.impl.hevm
```

is an exported package and is uses `com.swirlds.config.api.Configuration` in a [public signature here](https://github.com/hashgraph/hedera-services/blob/309d81511549804da5e7d4269942695ea5e4c66a/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/hevm/HederaEvmTransactionProcessor.java#L39).

-----

We don't know yet how we want to work with this check on the long run. If you prefer, I can revert this change as it is not needed for this branch. It should probably be addressed in the code by someone knowing this area instead of me (who has no idea about the code itself) changing dependencies.

Reply:
To illustrate this, I addressed what made `com.swirlds.common` transitive, as I am quite confident with that change:
https://github.com/hashgraph/hedera-services/commit/0c6f6d65e259eb83ab2ca028e78d5ce7afb7fa89

Reply:
I'm fine with the team reviewing and making changes to reduce the transitive requirements in a future PR.  Knowing that we have API leaks like this is sufficient clarification for now.